sdk/web-pubsub/web-pubsub-client/src/utils/abortablePromise.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortError, AbortSignalLike } from "@azure/abort-controller";
+
+export async function abortablePromise<T>(
+  promise: Promise<T>,
+  signal: AbortSignalLike
+): Promise<T> {
+  if (signal.aborted) {
+    throw new AbortError();
+  }
+
+  let onAbort: () => void;
+  // eslint-disable-next-line promise/param-names
+  const p = new Promise<T>((_, reject) => {
+    onAbort = (): void => {
+      reject(new AbortError());
+    };
+
+    signal.addEventListener("abort", onAbort);
+  });
+
+  try {
+    return await Promise.race([promise, p]);
```

Comment:
no need to await here, since you're just returning the Promise

```suggestion
    return Promise.race([promise, p]);
```

Reply:
Isn't await needed here so the promise resolves within the try/catch clause?

Reply:
Oh right, my mistake!